### PR TITLE
Typescript Port

### PR DESCRIPTION
**Pull request by bennett000**
This was moved from being a PR against master.  It has also been squashed down a bit

### CODE REVIEW

File: src/components/provider.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+interface INgRedux<T> extends Redux.Store<T> {
+  connect: IConnect
+}
+
+export function provider<T>(store: Redux.Store<T>) {
+  const _connector = new Connector(store);
+  const factory = (): INgRedux<T> => {
+    // TS doesn't seem to like this with ...spread :(
```
Comment (kosz) [inline src/components/provider.ts]: is this comment necessary ?

**Comment by kosz** [comment]
lgtm aside from one necessary squash remaining, and that one comment i made, which could be a non issue 

:shipit:

**Comment by bennett000** [comment]
@kosz I think I addressed all your concerns.

**Comment by wbuchwalter** [comment]
I don't have time to make a deep code review currently, but this looks really good!

**Pull request merged**